tests/Bonami/Collection/Monoid/MonoidTest.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Bonami\Collection\Monoid;
+
+use PHPUnit\Framework\TestCase;
+
+class MonoidTest extends TestCase
+{
+
+    /**
+     * @dataProvider provideFixtures
+     *
+     * @phpstan-template A
+     *
+     * @phpstan-param A $a
+     * @phpstan-param A $b
+     * @phpstan-param Monoid<A> $monoid
+     * @phpstan-param A $result
+     */
+    public function testFromMonoids($a, $b, Monoid $monoid, $result): void
+    {
+        self::assertSame($a, $monoid->concat($a, $monoid->getEmpty()));
+        self::assertSame($a, $monoid->concat($monoid->getEmpty(), $a));
+        self::assertSame($result, $monoid->concat($a, $b));
+    }
+
+    /** @phpstan-return iterable<array{0: mixed, 1: mixed, 2: Monoid<mixed>, 3: mixed}> */
+    public function provideFixtures(): iterable
```

Comment:
wow, it is possible to `yield` with return type `iterable`? I think `Generator` was necessary..

Reply:
It passes tests and phpstan does not complain ;-) We use similar trick in \Bonami\Collection\Map::getIterator where we return Traversable.

This works, because Generator is subtype of Traversable and subtype of pseudo type iterable.

Reply:
yeah every `Generator` is `iterable` but the opposite is not true..
Thats why I assumed, it would fail.